### ISSUE TITLE
Token dropout is set to Ture duing inference?

### ISSUE THREAD

**Issue by zhenyuhe00** [facebookresearch/esm#267]
Hi,
Congrats on your great works!
In your ESM-2 model code, I noticed that "token_dropout" is set to True. I wonder why is that.
Thanks in advance!

**Comment by nikitos9000** [comment]
Hi @zhenyuhe00 , that's just an implementation of Masked LM training, as in the paper `15% of amino acids are masked`, so the embeddings are scaled accordingly. I hope that answers your question

**Comment by spragud2** [comment]
I haven't come across scaling embeddings during inference for these types of models before. Is there a reference for doing that or is it something you guys did?

**Comment by tomsercu** [comment]
[Token drop](https://aclanthology.org/2020.coling-main.379.pdf) was introduced in our LMs in the ESM-1b paper (see supplement of PNAS paper Rives et al).
The idea behind it is that we're fully zero-ing out the embeddings instead of using a non-zero embedding for the `<mask>` token. 
Then the scaling logic is simply by direct analogy to [standard dropout](https://pytorch.org/docs/stable/generated/torch.nn.Dropout.html), which is motivated by having the same vector norm when dropout is turned on or off:
> Furthermore, the outputs are scaled by a factor of 1/(1−p)  during training. This means that during evaluation the module simply computes an identity function.

Overall token drop had a very minor effect. And as @nikitos9000 pointed out during inference there's no rescaling happening.

**Comment by zhenyuhe00** [comment]
Thanks a lot!

**Comment by jasperhyp** [comment]
Hi,  To follow up on this issue, I noticed in the appendix that not all 15% of excluded tokens are transformed into mask_idx.  Some are transformed into other tokens, and some are kept in their original form.  So these tokens are not dropped out, correct?

Also, the dropped out tokens would be the same as padding tokens, and so the final embedding would then be dependent on the bias term in each operation mainly?
